modules/core/src/main/java/org/apache/ignite/internal/processors/tracing/configuration/GridTracingConfiguration.java

```diff
@@ -0,0 +1,303 @@
+
+            return DEFAULT_CONFIGURATION_MAP;
+        }
+
+        if (metaStore == null) {
+            log.warning("Failed to retrieve tracing configuration — meta storage is not available.");
+
+            return DEFAULT_CONFIGURATION_MAP;
+        }
+
+        Map<TracingConfigurationCoordinates, TracingConfigurationParameters> res = new HashMap<>();
+
+        for (Scope scope : Scope.values()) {
+            String scopeSpecificKey = TRACING_CONFIGURATION_DISTRIBUTED_METASTORE_KEY_PREFIX + scope.name();
+
+            try {
+                for (Map.Entry<String, TracingConfigurationParameters> entry :
+                    ((Map<String, TracingConfigurationParameters>)metaStore.read(scopeSpecificKey)).entrySet()) {
+                    res.put(
+                        new TracingConfigurationCoordinates.Builder(scope).withLabel(entry.getKey()).build(),
+                        entry.getValue());
+                }
+            }
+            catch (IgniteCheckedException e) {
+                LT.warn(log, "Failed to retrieve tracing configuration");
+            }
+        }
+
+        return res;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean restoreDefaultConfiguration(@NotNull TracingConfigurationCoordinates coordinates) {
+        DistributedMetaStorage metaStore;
+
+        try {
+            metaStore = ctx.distributedMetastorage();
+        }
+        catch (Exception e) {
+            log.warning("Failed to restore tracing configuration for coordinates=[" + coordinates +
+                "] to default  — meta storage is not available.");
+
+            return false;
+        }
+
+        if (metaStore == null) {
+            log.warning("Failed to restore tracing configuration for coordinates=[" + coordinates +
+                "] to default  — meta storage is not available.");
+
+            return false;
+        }
+
+        String scopeSpecificKey = TRACING_CONFIGURATION_DISTRIBUTED_METASTORE_KEY_PREFIX + coordinates.scope().name();
+
+        boolean configurationSuccessfullyUpdated = false;
+
+        try {
+            while (!configurationSuccessfullyUpdated) {
+                HashMap<String, TracingConfigurationParameters> existingScopeSpecificTracingConfiguration =
+                    ctx.distributedMetastorage().read(scopeSpecificKey);
+
+                if (existingScopeSpecificTracingConfiguration == null) {
+                    // Nothing to do.
+                    return true;
+                }
+
+                HashMap<String, TracingConfigurationParameters> updatedScopeSpecificTracingConfiguration =
+                        new HashMap<>(existingScopeSpecificTracingConfiguration);
+
+                if (coordinates.label() != null)
+                    updatedScopeSpecificTracingConfiguration.remove(coordinates.label());
+                else
+                    updatedScopeSpecificTracingConfiguration.remove(null);
```

Comment:
Null mappings in Map always look suspicious.
Also, there's no simple way to drop all label-specific configurations in case we created lots of them.
I have a proposal. We can use two preset label names: DEFAULT (meaning that this configuration coordinate is applicable to all labels, for which label-specific configuration wasn't set) and ANY (meaning that this configuration coordinate is applicable to the whole scope).
DEFAULT will be stored as map key instead of null, ALL will be never present in configuration, but could be utilized for mass updates.
Examples:


        new TracingConfigurationCoordinates.Builder(Scope.TX).withLabel(DEFAULT).build();
        new TracingConfigurationCoordinates.Builder(Scope.TX).build();
        // ^ these two configuration coordinates are equal

        client.tracingConfiguration().addConfiguration(
            new TracingConfigurationCoordinates.Builder(Scope.TX).withLabel(ANY).build(),
            new TracingConfigurationParameters.Builder().withSamplingRate(0.5).build()
        );
        // ^ this code will remove all label-specific configurations and will set sampling rate for the whole TX scope to 0.5

        client.tracingConfiguration().restoreDefaultConfiguration(
            new TracingConfigurationCoordinates.Builder(Scope.TX).withLabel(ANY).build()
        );
        // ^ this code will restore defaults to the whole TX scope

        client.tracingConfiguration().retrieveConfiguration(
            new TracingConfigurationCoordinates.Builder(Scope.TX).withLabel(ANY).build()
        );
        // ^ this code will throw exception. "ANY" label can't be retrieved!

What do you think?
Probably, I'm over-engineering it, but I think need to wipe all label-specific configurations with a single command is likely to appear at least once.

Reply:
I believe that it's less intuitive and more complicated.

- Originally DEFAULT has different meaning - scope specific parameters that were predefined by us, not the user. So, using DEFAULT as scope specific,as you suggest, leads us to the necessity of introducing some default-default label.

- There's also a disadvantage in using predefined labels cause we actually change existing behavior, cause users may already have 'DEFAULT, ANY, ALL' labels for transactions.

- To be honest, I don't think that using null key is bad practice here. I believe that we might add some comment to clarify null's meaning as key.

- However I agree that it'll be useful to have an ability to drop all label-specific configurations for some scope, but I'd rather implement that with new api method.

All in all, it's discussable.